src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -153,6 +161,19 @@ public SCM build(SCMHead head, @CheckForNull SCMRevision revision) {
         return getGitSCMSource().build(head, revision);
     }
 
+    @NonNull
+    @Override
+    protected List<Action> retrieveActions(@NonNull SCMHead head, @CheckForNull SCMHeadEvent event,
+            @NonNull TaskListener listener) throws IOException, InterruptedException {       
+        List<Action> result = new ArrayList<>();
+        BitbucketDefaultBranch defaultBranch = scmHelper.getDefaultBranch(repository.getProjectName() , repository.getRepositoryName());
+        
+        if (StringUtils.equals(head.getName(), defaultBranch.getDisplayId())) {
```

Comment:
I haven't had the chance to test this, but I believe that we consumer every head event, not just those pertinent to a particular repository- I think we need to perform a check to ensure a given head event points to a repository as well.
The [branch source plugin](https://github.com/jenkinsci/bitbucket-branch-source-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java#L1012) solves this problem by attaching a metadata event on an `SCMSourceEvent` with the related repository, so a solution might look like:
```

    @NonNull
    @Override
    protected List<Action> retrieveActions(SCMSourceEvent event,
                                           @NonNull TaskListener listener) throws IOException, InterruptedException {
        return Collections.singletonList(new BitbucketRepoMetadataAction(repository));
    }

 @NonNull
    @Override
    protected List<Action> retrieveActions(@NonNull SCMHead head, SCMHeadEvent event,
                                           @NonNull TaskListener listener) throws IOException, InterruptedException {
        List<Action> actionList = new ArrayList<>();
        if (head instanceof GitBranchSCMHead && getDefaultBranchName().equals(((GitBranchSCMHead) head).getRef())) {
            SCMSourceOwner owner = getOwner();
            if (owner instanceof Actionable) {
                ((Actionable) owner).getActions(BitbucketRepoMetadataAction.class).stream()
                        .filter(action -> action.getRepository().equals(repository))
                        .findAny()
                        .ifPresent(/**Add action here**/);
    }
}

Reply:
I changed the implementation to be similar to the approach the branch source plugin is taking.